docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,499 @@
+|2                               |2             |0                             
+|**3 (recommended)**             |**2**         |**1**                             
+|4                               |3             |1                             
+|5                               |3             |2                             
+|6                               |4             |2                             
+|7                               |4             |3                             
+|=======================================================================================
+
+The minimum configuration size is `2 * cluster.master_nodes_failure_tolerance + 1`:
+
+[cols="<,<",options="header",]
+|====================================================================
+|`cluster.master_nodes_failure_tolerance` |Minimum configuration size
+|0                                        |1
+|**1**                                    |**3**
+|2                                        |5
+|3                                        |7
+|====================================================================
+
+It is permissible, but not recommended, to set
+`cluster.master_nodes_failure_tolerance` too low for your cluster or, put
+differently, to have more master-eligible nodes than the minimum configuration
+size. To do so is _safe_ in the sense that the cluster will not suffer from a
+split-brain however this setting is configured, but if
+`cluster.master_nodes_failure_tolerance` is too low then your cluster may not
+tolerate as many failures as expected.
+
+[float]
+==== Even numbers of master-eligible nodes
+
+There should normally be an odd number of master-eligible nodes in a cluster.
+If there is an even number then Elasticsearch will put all but one of them into
+the configuration to ensure that the configuration has an odd size. This does
+not decrease the failure-tolerance of the cluster, and in fact improves it
+slightly: if the cluster is partitioned into two even halves then one of the
+halves will contain a majority of the masters and will be able to keep
+operating, whereas if all of the master-eligible nodes' votes were counted then
+neither side could make any progress in this situation.
+
+[float]
+==== Adding master-eligible nodes
+
+It is recommended to have a small, fixed, number of master-eligible nodes in a
+cluster, and to scale the cluster up and down by adding and removing
+non-master-eligible nodes only. However there are situations in which it may be
+necessary to add some master-eligible nodes to a cluster, such as when
+migrating a cluster onto a new set of nodes without downtime.
+
+If you wish to add some master-eligible nodes to your cluster, simply configure
+the new nodes to find the existing cluster and start them up. Once the new
+nodes have joined the cluster, you may be able to increase the
+`cluster.master_nodes_failure_tolerance` setting to match. (TODO do we log
+info/warn messages about this?)
```

Comment:
@DaveCTurner I think it makes sense in addition to adding nodes to the cluster and removing nodes from cluster add a separate paragraph on replacing nodes in the cluster. Of course, it's similar to adding/removing nodes, but in this case, there is no need to touch `cluster.master_nodes_failure_tolerance` settings during replacement. Unfortunately, the way we describe this setting is a bit confusing, so it would be better to have a separate paragraph that expains that there is no need to touch it if you do simple node replacement.

Reply:
This is a good point and I have added a section on "maintenance" giving general guidance about this.